examples/tv-casting-app/tv-casting-common/core/CastingPlayerDiscovery.cpp

```diff
@@ -63,16 +63,59 @@ CHIP_ERROR CastingPlayerDiscovery::StartDiscovery(uint32_t deviceTypeFilter)
 
 CHIP_ERROR CastingPlayerDiscovery::StopDiscovery()
 {
-    ChipLogProgress(Discovery, "CastingPlayerDiscovery::StopDiscovery() called");
+    ChipLogProgress(Discovery, "CastingPlayerDiscovery::StopDiscovery() mCastingPlayers: %lu, mCastingPlayersInternal: %lu",
+                    mCastingPlayers.size(), mCastingPlayersInternal.size());
     VerifyOrReturnError(mState == DISCOVERY_RUNNING, CHIP_ERROR_INCORRECT_STATE);
     ReturnErrorOnFailure(mCommissionableNodeController.StopDiscovery());
 
+    // Clear mCastingPlayersInternal of disconnected CastingPlayers
+    ClearDisconnectedCastingPlayersInternal();
+    for (const auto & player : mCastingPlayers)
+    {
+        mCastingPlayersInternal.push_back(player);
+    }
```

Comment:
Why not just use the copy contructor to copy the vector?

Reply:
Since I was keeping the connected players in mCastingPlayersInternal before copying. Copying mCastingPlayers to mCastingPlayersInternal would delete the players in mCastingPlayersInternal. Thinking about this further, there is no need to keep the the connected players in mCastingPlayersInternal when stopDiscovery() is called. I'll change this to the following:
```
    // Copy mCastingPlayers to mCastingPlayersInternal
    mCastingPlayersInternal = std::vector<memory::Strong<CastingPlayer>>(mCastingPlayers);
```